eyeball/src/shared.rs

```diff
@@ -232,6 +257,30 @@ impl<T> Drop for Observable<T> {
     }
 }
 
+/// A weak reference to a shared [`Observable`].
+///
+/// This type is only useful in niche cases, since one generally shouldn't nest
+/// interior-mutable types in observables, which includes observables
+/// themselves.
+///
+/// See [`std::sync::Weak`] for a general explanation of weak references.
+#[derive(Debug)]
+pub struct WeakObservable<T> {
+    state: Weak<RwLock<ObservableState<T>>>,
```

Comment:
How problematic would it be to have `Clone` for the `Weak` one as well? I guess you would need to upgrade every time you clone?

Reply:
That was intended to be there. It doesn't need to upgrade every time.

Reply:
Ah great, thanks.